### PR TITLE
Fix daily runner branch drift before PR creation

### PR DESCRIPTION
**Pull request by hushline-dev**
## What changed
- add self-healing in the daily runner when worktree state drifts off the issue branch before commit
- move a stray commit back onto the issue branch if it lands on `main`, then repair `main` back to `origin/main`
- block PR creation only when the issue branch truly has no commits ahead of its base
- add regression tests covering branch reattachment, stray-commit recovery, and empty PR branch rejection

## Why
A recent daily runner execution pushed `codex/daily-issue-1777`, then failed to open a PR because GitHub saw no commits between `main` and that branch. The work commit existed locally but had landed on `main` instead of the issue branch. This change makes the runner recover that state so it can finish cleanly instead of pushing a stale branch ref and failing at PR creation.

## Validation
- `make test`
- `make lint`

## Manual testing
- not applicable; runner behavior is covered by automated shell-script regression tests and full repo validation

## Risks / follow-ups
- the recovery path force-moves the expected issue branch to the new `HEAD` commit when branch drift is detected; that is intentional, but it assumes the current `HEAD` is the runner's newly created work commit

### CODE REVIEW

File: scripts/agent_daily_issue_runner.sh [hunk 1]
```diff
@@ -2068,9 +2157,13 @@ main() {
   fi
   git commit -m "$COMMIT_MESSAGE"
 
+  ensure_head_commit_on_branch "$BRANCH_NAME" "$BASE_BRANCH"
+  require_branch_has_unique_commits "$PR_BASE_BRANCH" "$BRANCH_NAME"
```
[the remark this answers is not in the document]
Comment (hushline-dev) [reply scripts/agent_daily_issue_runner.sh]: Addressed in `ecde6dcd`.

The runner now tracks a concrete compare ref separately from the PR base branch name, so child-issue uniqueness checks use `origin/<epic>` when the epic branch only exists as a fetched remote ref. I also added a regression test that exercises the existing-epic path and asserts the `git rev-list` comparison uses `origin/codex/epic-1735..codex/daily-issue-1732`.

Validation rerun:
- `make test TESTS="tests/test_agent_daily_issue_runner.py"`
- `make lint`
